File: TaskListAngularLab/ClientApp/src/app/Task.ts
export interface Task{
    id: number;
    tmName: string;
    taskName: string;
    taskDescription: string;
    dueDate: Date;
    isCompleted: boolean;
}

export class Convert {
    public static toTask(json: string): Task {
        return JSON.parse(json);
    }

    public static taskToJson(value: Task): string {
        return JSON.stringify(value);
    }
}

File: TaskListAngularLab/Models/Task.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace TaskListAngularLab.Models
{
    public class Task
    {
        [Key]
        public int Id { get; set; }
        public string TmName { get; set; }
        public string TaskName { get; set; }
        public string TaskDescription { get; set; }
        public DateTime DueDate { get; set; }
        public bool isCompleted { get; set; }
    }
}
